src/libawkward/Content.cpp

```diff
@@ -1275,6 +1306,11 @@ namespace awkward {
                                   "with NumPy-style advanced indexing");
     }
 
+    // would length ever not be 1 when the content is jagged?
+    if ( length() == 1 && dynamic_cast<SliceJagged64*>(missing.content().get())) {
+      return getitem_next_missing_jagged(missing, tail, advanced, shallow_copy());
+    }
+
     ContentPtr tmp = check_missing_jagged(shallow_copy(), missing);
```

Comment:
Is `check_missing_jagged` relevant anymore? If these if statements (checking all option-type layout nodes) now have zero coverage in the test suite, then they're dead code and can be removed (maybe pulling the only non-option-type test out):

https://github.com/scikit-hep/awkward-1.0/blob/bb205f3ff9f5db5c131ff53ec85cd3ebac62a1e8/src/libawkward/Content.cpp#L1226-L1257

Reply:
I think this new behavior does supersede that whole section, but will check.

Reply:
None of the unit tests reached that code after this change, so I deleted it. I think it makes sense too.